text/0000-deprecate-ember-k.md

```diff
@@ -0,0 +1,105 @@
+- Start Date: 2016-11-18
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+The `Ember.K` utility function is a low level utility that has lost most of its value today,
+like the wisdom tooth of the frameworks were we only think of it when it gives us problems.
+
+# Motivation
+
+Let's start explaining what is `Ember.K`.
+
+It is this function:
+
+```js
+Ember.K = function() {
+  return this;
+}
+```
+
+The purpose of this utility was to avoid some amount of boilerplate code
+and limit the creation of function instances in Ember's internals.
+
+In a world of globals, effectively writing `somefn: Ember.K` was shorter
+than writing
+
+```js
+someFn: function() {
+  return this;
+}
+```
+and generated less function alocations.
+
+However with the introduction of ES6 modules and the modularization of Ember
+in process (#176), keeping this feature would require to design an import path for it.
+
+While doable, any possible benefit in code size or allocations obtained by reusing
+the exact same Function instance everywhere would be greatly smashed by the
+overhead both in space and CPU cycles of importing it in AMD-transpiled modules.
+It's worth noting that ES6 shorthand method syntax has made writing functions
+less verbose and those functions gain an internal name property that shows in
+stack traces making debugging easier.
+
+The second downside of reusing the same instance in many places is that if for
+some reason the VM deoptimizes that function, that deoptimization is spreaded
+across all the usages of `Ember.K`.
```

Comment:
@rwjblue mentioned it in slack and [here](https://github.com/emberjs/rfcs/pull/176#discussion_r88422065) an some smart fellows 👍 'd it